recipes/newrelic/infrastructure/super-agent/debian.yml

```diff
@@ -423,21 +490,34 @@ install:
             sed -i 's/\(endpoint: https:\/\/opamp.\).*/\1'"service.newrelic.com\/v1\/opamp"'/' /etc/newrelic-super-agent/config.yaml
           fi
         - |
+          if [ -f /etc/newrelic-super-agent/.nr-cli ] ; then
+            exit 0
+          fi
           if [ "{{.NEW_RELIC_SUPER_AGENT_FLEET_ENABLED}}" != "false" ] ; then
             sed -i 's/api-key: API_KEY_HERE/api-key: {{.NEW_RELIC_LICENSE_KEY}}/g' /etc/newrelic-super-agent/config.yaml
           fi
 
     config_host_monitoring:
       cmds:
         - |
+          # Log if a previous execution of this recipe was detected, indicating we are skipping this task
+          if [ -f /etc/newrelic-super-agent/.nr-cli ] ; then
+            echo "A previous execution of this recipe was detected. Skipping "config_host_monitoring" task."
+            exit 0
+          fi
           if [ "{{.NEW_RELIC_SUPER_AGENT_HOST_MONITORING_SOURCE}}" = "otel" ]; then
             echo 'is_integrations_only: true' >> /etc/newrelic-infra.yml
             mkdir -p /etc/newrelic-super-agent/fleet/agents.d/nr-otel-collector/values
             cp /etc/newrelic-super-agent/examples/values-nr-otel-collector-agent-linux-0.1.0.yaml /etc/newrelic-super-agent/fleet/agents.d/nr-otel-collector/values/values.yaml
           else
             if [ "{{.NR_CLI_NRDOT}}" != "false" ]; then
               mkdir -p /etc/newrelic-super-agent/fleet/agents.d/nr-otel-collector/values
-              cp /etc/newrelic-super-agent/examples/values-nr-otel-collector-gateway-0.1.0.yaml /etc/newrelic-super-agent/fleet/agents.d/nr-otel-collector/values/values.yaml
+              cp /etc/newrelic-super-agent/examples/values-nr-otel-collector-agent-linux-0.1.0.yaml /etc/newrelic-super-agent/fleet/agents.d/nr-otel-collector/values/values.yaml
+            else
+              if [ "{{.NR_CLI_NRDOT}}" != "false" ]; then
+                mkdir -p /etc/newrelic-super-agent/fleet/agents.d/nr-otel-collector/values
+                cp /etc/newrelic-super-agent/examples/values-nr-otel-collector-gateway-0.1.0.yaml /etc/newrelic-super-agent/fleet/agents.d/nr-otel-collector/values/values.yaml
+              fi
             fi
```

Comment:
This code doesn't seem correct. The second `if` will never happen. A greater mind would explain like this:

> The code contains a logical error in the else block. The condition inside the else block is the same as the condition in the if block, which means the code inside the else block will never be executed.

Reply:
that's true; thanks for spotting this @rubenruizdegauna :)

Reply:
Fixed in https://github.com/newrelic/open-install-library/pull/1094/commits/0b5d29dad079f86e4d978740823a25ae1c445982!